www/js/control/uploadService.ts

```diff
@@ -0,0 +1,131 @@
+ */
+async function getUploadConfig() {
+    return new Promise<string[]>(async function (resolve, reject) {
+        logInfo( "About to get email config");
+        let url = [];
+        try {
+            let response = await fetch("json/uploadConfig.json");
+            let uploadConfig = await response.json();
+            logDebug("uploadConfigString = " + JSON.stringify(uploadConfig['url']));
+            url.push(uploadConfig["url"]);
+            resolve(url);
+        } catch (err) {
+            try{
+                let response = await fetch("json/uploadConfig.json.sample");
+                let uploadConfig = await response.json();
+                logDebug("default uploadConfigString = " + JSON.stringify(uploadConfig['url']));
+                console.log("default uploadConfigString = " + JSON.stringify(uploadConfig['url']))
+                url.push(uploadConfig["url"]);
+                resolve(url);
+            } catch (err) {
+                displayError(err, "Error while reading default upload config");
+                reject(err);
+            }
+        } 
+    })
+}
+
+function onReadError(err) {
+    displayError(err, "Error while reading log");
+}
+
+function onUploadError(err) {
+    displayError(err, "Error while uploading log");
+}
+
+function readDBFile(parentDir, database, callbackFn) {
+    return new Promise(function(resolve, reject) {
+        window['resolveLocalFileSystemURL'](parentDir, function(fs) {
+            console.log("resolving file system as ", fs);
+            fs.filesystem.root.getFile(fs.fullPath+database, null, (fileEntry) => {
+                console.log(fileEntry);
+                fileEntry.file(function(file) {
+                  console.log(file);
+                  var reader = new FileReader();
+
+                  reader.onprogress = function(report) {
+                    console.log("Current progress is "+JSON.stringify(report));
+                    if (callbackFn != undefined) {
+                        callbackFn(report.loaded * 100 / report.total);
+                    }
+                  }
+
+                  reader.onerror = function(error) {
+                    console.log(this.error);
+                    reject({"error": {"message": this.error}});
+                  }
+
+                  reader.onload = function() {
+                    console.log("Successful file read with " + this.result['byteLength'] +" characters");
+                    resolve(new DataView(this.result as ArrayBuffer));
+                  }
+
+                  reader.readAsArrayBuffer(file);
+                }, reject);
+            }, reject);
+        });
+    });
+}
+
+const sendToServer = function upload(url, binArray, params) {
+    //this was the best way I could find to contact the database, 
+    //had to modify the way it gets handled on the other side
+    //the original way it could not find "reason"
+    const urlParams = "?reason=" + params.reason + "&tz=" + params.tz;
```

Comment:
This is an old comment, I will remove it. It took me a long time to figure out that I just needed to use url encoding because that's what the old method was doing under the hood. Once I figured out the right way to do it I was able to reset the code on [`em bug upload`](https://github.com/e-mission/em-bug-upload).